### PR TITLE
fix(ngcc): handle minified ES5 code

### PR DESCRIPTION
**Pull request by petebacondarwin**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by IgorMinar** [comment]
can you please provide more info in the commit message?

**Comment by petebacondarwin** [comment]
Yes this is still Draft

File: packages/compiler-cli/ngcc/src/host/esm2015_host.ts [hunk 1]
```diff
@@ -1104,7 +1104,8 @@ export class Esm2015ReflectionHost extends TypeScriptReflectionHost implements N
    * does not match.
    */
   protected getHelperCall(statement: ts.Statement, helperNames: string[]): ts.CallExpression|null {
-    if (ts.isExpressionStatement(statement)) {
+    if ((ts.isExpressionStatement(statement) || ts.isReturnStatement(statement)) &&
+        statement.expression) {
```
Comment (gkalpak) [inline packages/compiler-cli/ngcc/src/host/esm2015_host.ts]: Nit: We only need to check `statement.expression` for `ReturnStatement`s :speak_no_evil:

**Review by gkalpak — APPROVED** [review]
:+1:
Commit message typo: `statemetn` --> `statement`

File: packages/compiler-cli/ngcc/src/host/esm2015_host.ts (hunk 1 above)
Comment (petebacondarwin) [reply packages/compiler-cli/ngcc/src/host/esm2015_host.ts]: That may be true but TS is not clever enough to narrow `statement` to `ExpressionStatement|(ReturnStatement&{expression:Expression)` by itself.

Comment (gkalpak) [reply packages/compiler-cli/ngcc/src/host/esm2015_host.ts]: 😞

**Comment by dioslaska** [comment]
@petebacondarwin  Does this also supposed to make ngcc work with renamed variables? It still fails for me with the latest rc3 version. Example with a simple test module and component:

Fails:

```
    import { Component as n, NgModule as t } from "@angular/core";
    import { CommonModule as r } from "@angular/common";

    var o = function() {
      function t() {}
      t.decorators = [ {
        type: n,
        args: [ {
          selector: "test-input",
          template: "<label><ng-content></ng-content><input /></label>"
        } ]
      } ];
      return t;
    }();

    var e = function() {
      function n() {}
      n.decorators = [ {
        type: t,
        args: [ {
          imports: [ r ],
          declarations: [ o ],
          exports: [ o ]
        } ]
      } ];
      return n;
    }();

    export { o as TestComponent, e as TestModule };
```

Works:

```
    import { Component, NgModule } from '@angular/core';
    import { CommonModule } from '@angular/common';

    var TestComponent = /** @class */ (function () {
        function TestComponent() {
        }
        TestComponent.decorators = [
            { type: Component, args: [{
                        selector: 'test-input',
                        template: "<label><ng-content></ng-content><input /></label>"
                    }] }
        ];
        return TestComponent;
    }());

    var TestModule = /** @class */ (function () {
        function TestModule() {
        }
        TestModule.decorators = [
            { type: NgModule, args: [{
                        imports: [CommonModule],
                        declarations: [TestComponent],
                        exports: [TestComponent]
                    },] }
        ];
        return TestModule;
    }());

    export { TestComponent, TestModule };
```

**Comment by dioslaska** [comment]
Created a reproduction repo with the minified fesm5 file: https://github.com/acidb/mobiscroll-ivy-test/tree/test

**Comment by petebacondarwin** [comment]
@dioslaska  - please open a new ticket for this issue rather than tacking it on to a closed pull request. This will make it easier to deal with the problem and track the work required.

I think the problem you are experiencing is that ngcc does not appear cope with renames at the point of export:

```ts
export { o as TestComponent, e as TestModule };
```

**Pull request closed without merging**